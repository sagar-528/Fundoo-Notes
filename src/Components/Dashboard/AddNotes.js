import React, { Component } from 'react'
import NoteCss from '../../Styles/NoteCss'
import {View, ScrollView, TextInput} from 'react-native'
import { Appbar } from 'react-native-paper'

export class AddNotes extends Component {

constructor(props) {
    super(props)

    this.state = {
        title : '',
        notes : '' 
    }
}

handleTitle = (title) => {
 this.setState({
        title : title
    })
}

handleNotes = (notes) => {
 this.setState({
        notes : notes
    })
}

    render() {
        return (
            <View style = {NoteCss.mainContainer}>
            <View>
                <Appbar style = {NoteCss.header_style}>
                    <Appbar.Action 
                        style = {{marginLeft : 10}}
                        icon = 'keyboard-backspace'/>
                    <Appbar.Content />
                    <Appbar.Action
                        style = {NoteCss.header_icon_style}                             
                        icon = 'pin-outline'/>
                    <Appbar.Action    
                        style = {NoteCss.header_icon_style}                          
                        icon = 'bell-plus-outline'/>
                    <Appbar.Action 
                        icon = 'archive-arrow-down-outline'/>
                </Appbar>
            </View>
            <ScrollView style = {{marginBottom : 60}}> 
                <TextInput
                    style = {NoteCss.title_style}
                    multiline = {true} 
                    placeholder = 'Title'
                    onChangeText = {this.handleTitle}
                />
                <TextInput
                    style = {NoteCss.note_style}
                    multiline = {true} 
                    placeholder = 'Note'
                    onChangeText = {this.handleNotes}
                />
            </ScrollView>
            <View style = {NoteCss.bottom_view}>
                <Appbar style = {NoteCss.bottom_appbar_style}>
                    <Appbar.Action 
                        icon = 'plus-box-outline'/>
                    <Appbar.Content/>
                    <Appbar.Action 
                        icon = 'undo-variant'/>
                    <Appbar.Action 
                        icon = 'redo-variant'/>
                    <Appbar.Content/>
                    <Appbar.Action 
                        icon = 'dots-vertical'/>
                </Appbar>
            </View>
        </View>
        )
    }
}

export default AddNotes
