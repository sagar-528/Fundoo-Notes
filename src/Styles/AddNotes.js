import { StyleSheet } from 'react-native';

const AddNotesStyle = StyleSheet.create({

    mainContainer : {
        flex : 1,
        backgroundColor : 'white'
      },

      header_style : {
        backgroundColor : 'transparent'
      },

      header_icon_style : {
        marginRight : 10
      },

      title_style : {
        marginLeft : 20,
        marginRight : 20,
        marginTop : 20,
        fontSize : 22,
      },
    
      note_style : {
        marginLeft : 20,
        marginRight : 20,
        fontSize : 18,
      },

      bottom_view : {
        width: '100%',
        position: 'absolute',
        bottom : 0,
      },

      bottom_appbar_style : {
        backgroundColor : 'white' 
      },
})

export default AddNotesStyle;