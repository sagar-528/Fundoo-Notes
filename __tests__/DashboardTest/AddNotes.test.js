import React from 'react';
import {shallow} from 'enzyme';
import AddNotes from '../../src/Components/Dashboard/AddNotes'

describe('test AddNotes', () => {
    
    it('should match to snapshot', () => {
        const component = shallow(<AddNotes />)
        expect(component).toMatchSnapshot();
    })
})