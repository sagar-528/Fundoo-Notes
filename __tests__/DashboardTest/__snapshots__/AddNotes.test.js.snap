// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AddNotes should match to snapshot 1`] = `ShallowWrapper {}`;
